### PR TITLE
change crypto(clib) to sha1(lualib)

### PR DESCRIPTION
**Pull request by echohes**
luacrypto is deprecated

### CODE REVIEW

File: lua-tarantool-scm-1.rockspec
Comment (Totktonada) [inline lua-tarantool-scm-1.rockspec]: This change does not look related to others. Can you clarify it and, if it is really necessary, move to its own commit?

File: tarantool.moon
Comment (Totktonada) [inline tarantool.moon]: AFAIS `sha1.binary` is in the https://github.com/mpeterv/sha1 module from the initial commit, so I don't see a point to check it for existence.

Comment (echohes) [reply tarantool.moon]: I also thought about it, but decided to leave it for correctness (with the old logic).

File: lua-tarantool-scm-1.rockspec
Comment (echohes) [reply lua-tarantool-scm-1.rockspec]: So far I have returned it back, but I tested several versions, and it worked.

**Review by echohes — COMMENTED** [review]
So far I have returned the bind to version 5.1

File: tarantool.moon
Comment (Totktonada) [reply tarantool.moon]: It would make sense if we would fallback to `require('crypto')`, but we don't. As for me, the fallback is not necessary. I don't even think that we really bother about performance here, because we need to calculate a hash function once for a connection: during authentication.

File: lua-tarantool-scm-1.rockspec
Comment (Totktonada) [reply lua-tarantool-scm-1.rockspec]: Okay. Please, squash commits and force-push.

File: tarantool.moon
Comment (echohes) [reply tarantool.moon]: Okay, then I'll remove this check.

**Review by echohes — COMMENTED** [review]
removed unnecessary

**Comment by ligurio** [comment]
@echohes Vadim, could you set checkbox "Allow edits and access to secrets by maintainers" in pull request?
I want to polish changes before merging.

**Comment by ligurio** [comment]
> @echohes Vadim, could you set checkbox "Allow edits and access to secrets by maintainers" in pull request?
> I want to polish changes before merging.

Squashed patches to a single commit and updated commit message. Thanks!

**Review by ligurio — APPROVED** [review]
LGTM

**Review by Totktonada — APPROVED** [review]

**Pull request merged**
